tests/test_open_options.py

```diff
@@ -0,0 +1,32 @@
+"""Tests of dataset opening options and driver choice"""
+
+from packaging.version import parse
+import pytest
+
+import rasterio
+from rasterio.transform import Affine
+
+
+@pytest.mark.xfail(parse(rasterio.__gdal_version__) < parse('2.0'),
+                   reason="Requires a GDAL 2.0 feature")
+def test_fail_with_missing_driver():
+    """Fail to open a GeoTIFF without the GTiff driver"""
+    with pytest.raises(rasterio.errors.RasterioIOError):
+        rasterio.open('tests/data/RGB.byte.tif', driver='BMP')
```

Comment:
We can't open a GeoTIFF using the BMP driver.

Reply:
Could this test also probe the exception message to ensure the error is triggered by an incompatible file + driver pairing?

Reply:
No, it's only a "'tests/data/RGB.byte.tif' not recognized as a format" message.